pkg/pillar/cmd/monitor/subscriptions.go

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package monitor
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+func handlePhysicalIOAdapterCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handlePhysicalIOAdapterUpdate(ctxArg, statusArg)
+}
+
+func handlePhysicalIOAdapterModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handlePhysicalIOAdapterUpdate(ctxArg, statusArg)
+}
+
+func handlePhysicalIOAdapterUpdate(ctxArg interface{}, statusArg interface{}) {
+	ctx := ctxArg.(*monitor)
+	status := statusArg.(types.PhysicalIOAdapterList)
+	ctx.IPCServer.sendIpcMessage("IOAdapters", status)
+}
+
+func handlePhysicalIOAdapterDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+}
+
+func handleNetworkStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleNetworStatusUpdate(statusArg, ctxArg)
+
+}
+func handleNetworkStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+}
+
+func handleNetworkStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, _ interface{}) {
+	handleNetworStatusUpdate(statusArg, ctxArg)
+}
+
+func handleNetworStatusUpdate(statusArg interface{}, ctxArg interface{}) {
+	status := statusArg.(types.DeviceNetworkStatus)
+	ctx := ctxArg.(*monitor)
+	ctx.IPCServer.sendIpcMessage("NetworkStatus", status)
+}
+
+func handleDownloaderStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleDownloaderStatusUpdate(statusArg, ctxArg)
+}
+
+func handleDownloaderStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, _ interface{}) {
+	handleDownloaderStatusUpdate(statusArg, ctxArg)
+}
+
+func handleDownloaderStatusUpdate(statusArg interface{}, ctxArg interface{}) {
+	status := statusArg.(types.DownloaderStatus)
+	ctx := ctxArg.(*monitor)
+	ctx.IPCServer.sendIpcMessage("DownloaderStatus", status)
+}
+
+func handleDownloaderStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
```

Comment:
What is this function for?

Reply:
> What is this function for?

@christoph-zededa It seems I have to handle it to inform UI that downloading is over. It is empty for now because I do not display the status (only in debug tab which is not visible in production build).

Reply:
ah okay, so this is wip?

Reply:
@christoph-zededa yes, it is a WIP and will be fixed in coming PRs